elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -122,15 +146,61 @@ public void commit() {
         criterion = CriterionFilterOperation.andWithNull(criterion,
                 criterionFilterOperation.applyAll(filteredPredicates));
 
-        // if no criterion then return all objects
-        if (criterion == null) {
-            return loadObjects(loadClass);
+
+        final Pagination pagination = filterScope.hasPagination() ? filterScope.getRequestScope().getPagination()
+                : null;
+
+        // if we have sorting and sorting isn't empty, then we should pull dictionary to validate the sorting rules
+        Set<Order> validatedSortingRules = null;
+        if (filterScope.hasSortingRules()) {
+            final Sorting sorting = filterScope.getRequestScope().getSorting();
+            final Map<String, Sorting.SortOrder> validSortingRules = sorting.getValidSortingRules(loadClass,
+                    filterScope.getRequestScope().getDictionary());
+            if (!validSortingRules.isEmpty()) {
+                final Set<Order> sortingRules = new LinkedHashSet<>();
+                validSortingRules.entrySet().stream().forEachOrdered(entry ->
+                        sortingRules.add(entry.getValue().equals(Sorting.SortOrder.desc)
+                                ? Order.desc(entry.getKey())
+                                : Order.asc(entry.getKey()))
+                );
+                validatedSortingRules = sortingRules;
+            }
+        }
+
+        return loadObjects(loadClass, Optional.ofNullable(criterion), Optional.ofNullable(validatedSortingRules),
+                Optional.ofNullable(pagination));
+    }
+
+    /**
+     * Generates the Hibernate ScrollableIterator for Hibernate Query.
+     * @param loadClass The hibernate class to build the query off of.
+     * @param criterion The Optional criterion object.
+     * @param sortingRules The possibly empty sorting rules.
+     * @param pagination The Optional pagination object.
+     * @param <T> The return Iterable type.
+     * @return The Iterable for Hibernate.
+     */
+    public <T> Iterable<T> loadObjects(final Class<T> loadClass, final Optional<Criterion> criterion,
```

Comment:
This method could literally be the only loadObjects code. However at the end of the day, outside of GET requests, this would be generating garbage in terms of Optional.empty(), so I kept the other overloads.